lib/tools/adb-commands.js

```diff
@@ -1142,17 +1142,29 @@ methods.removeLogcatListener = function (listener) {
  */
 methods.getPIDsByName = async function (name) {
   log.debug(`Getting IDs of all '${name}' processes`);
-  if (!_.isBoolean(this._isPidofAvailable)) {
-    this._isPidofAvailable = parseInt(await this.shell(['pidof --help > /dev/null; echo $?']), 10) === 0;
+  if (!_.isBoolean(this._isPgrepAvailable)) {
+    // pgrep is in priority, since pidof has been reported of having bugs on some platforms
+    const pgrepOutput = _.trim(await this.shell(['pgrep --help; echo $?']));
+    this._isPgrepAvailable = parseInt(_.last(pgrepOutput.split(/\s+/)), 10) === 0;
+    if (this._isPgrepAvailable) {
+      this._canPgrepUseFullCmdLineSearch = pgrepOutput.test(/^-f\b/m);
+    } else {
+      this._isPidofAvailable = parseInt(await this.shell(['pidof --help > /dev/null; echo $?']), 10) === 0;
```

Comment:
Is this more performant that `which pidof`?

Reply:
this works even in case if `which` is not available on the target platform. Performance-wise I would say there is no difference.